x-pack/packages/kbn-entities-schema/src/schema/common.ts

```diff
@@ -145,7 +145,7 @@ export type MetadataField = z.infer<typeof metadataSchema>;
 export const identityFieldsSchema = z
   .object({
     field: z.string(),
-    optional: z.boolean(),
+    optional: z.literal(false),
```

Comment:
This blocks people from using optional fields even though technically this is still supported in the code, and we transform the existing saved objects to remove them.

This is mainly due to not being able to change the saved object mapping.